internal/access/credential.go

```diff
@@ -225,3 +234,43 @@ func checkPasswordRequirements(db data.ReadTxn, password string) error {
 
 	return nil
 }
+
+// checkPasswordBreached checks if the password is found in previous data breaches. It uses
+// Have I Been Pwned's API to perform this check. The API takes the first 5-characters prefix
+// of the SHA1 checksum of the password and returns a carriage returned list. A password is
+// breached if the remaining characters of its SHA1 checksum appears in this list.
+// Ref: https://haveibeenpwned.com/API/v3#PwnedPasswords
+func checkPasswordBreached(password string) error {
+	//nolint:gosec // this is required for the API
+	sum := sha1.Sum([]byte(password))
+	hexsum := hex.EncodeToString(sum[:])
+
+	request, err := http.NewRequestWithContext(context.Background(), http.MethodGet, fmt.Sprintf("https://api.pwnedpasswords.com/range/%s", hexsum[:5]), nil)
+	if err != nil {
+		return err
+	}
+
+	response, err := http.DefaultClient.Do(request)
```

Comment:
I guess we need some kind of timeout on this, probably set on the context.  The default client has no timeout set. We probably don't want to wait any more than 5-10 seconds, right?

I'm not sure if we should ignore the error on timeout (allowing someone to change their password when we can't check it), or if we should fail the request in that case. My guess is ignore the timeout error would be best.